### PR TITLE
Postview header

### PR DESCRIPTION
**Pull request by Ali-Khamis**
added post view header component as needed in this [issue](https://github.com/m3ntorship/pickify-frontend/issues/118)

### CODE REVIEW

File: modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import UserInfo from '../../UserInfo/Index';
+import * as EDivider from '../../../atoms/Divider/types/EDivider';
+
+const PostViewHeader: FC<PostViewHeaderTypes.IProps> = ({
+  size = 'medium',
+  variant,
+  imgSrc,
+  name,
+  date,
+  onClick,
```
Comment (Nilegfx) [inline modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx:15]: please be more explicit. `onClick`-ing what exactly?
I would suggest `onClickOption` or something along the lines.

File: modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.test.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+describe('Testing postViewHeader with snapshot', () => {
+  it('if variant equalls annonymous, name becomes Anonymous as well even if a name is set', () => {
+    const tree = renderer
+      .create(
+        <PostViewHeader
+          id="test"
+          size="medium"
+          variant="anonymous"
+          date="2 hours"
+          name="Marzouk el akta3"
```
Comment (Nilegfx) [inline modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.test.tsx:17]: 😂😂😂😂

File: modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.test.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      <PostViewHeader
+        id="test"
+        size="medium"
+        variant="anonymous"
+        date="2 hours"
+        name="Marzouk el akta3"
+        onClick={handleClick}
+      />,
+    );
+    const editIcon: TargetElement = screen.getByTestId('test');
```
Comment (Nilegfx) [inline modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.test.tsx:40]: `postOptionsIcon`

File: modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.test.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    userEvent.click(editIcon);
+    expect(handleClick).toBeCalled();
```
Comment (Nilegfx) [inline modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.test.tsx:42]: better to use `toBeCalledTimes(1)`

File: modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeaderTypes.d.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import type { IUserInfo } from '../../UserInfo/IUserInfo';
+
+declare namespace PostViewHeaderTypes {
+  export interface IProps extends IUserInfo.IProps {
```
Comment (Nilegfx) [inline modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeaderTypes.d.ts:4]: why didn't you add the rest of props here?

File: modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import type { PostViewHeaderTypes } from './PostViewHeaderTypes';
+import UserInfo from '../../UserInfo/Index';
+import * as EDivider from '../../../atoms/Divider/types/EDivider';
+
+const PostViewHeader: FC<PostViewHeaderTypes.IProps> = ({
+  size = 'medium',
+  variant,
+  imgSrc,
+  name,
+  date,
```
Comment (Nilegfx) [inline modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx:14]: which component responsible to parse/transform the date from iso formate to a human readable format like "2 days" or "3 hours"?

**Review by Nilegfx — CHANGES_REQUESTED** [review]
some changes requested.

File: modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeaderTypes.d.ts (hunk 5 above)
Comment (Ali-Khamis) [reply modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeaderTypes.d.ts:4]: it already takes the props from the IUserInfo

File: modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx (hunk 6 above)
Comment (Ali-Khamis) [reply modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx:14]: we decided to parse the date in the parent

File: modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeaderTypes.d.ts (hunk 5 above)
Comment (Nilegfx) [reply modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeaderTypes.d.ts:4]: I am from mobile and I didn't see the extends 😂

File: modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx (hunk 6 above)
Comment (Nilegfx) [reply modules/shared/components/molecules/PostViewHeader/UserInfo/Index.tsx:14]: I wouldn't suggest so. I would vote for the userinfo to receive date and parse it. I belive the parent will have too much responsibilities if we started to delegate everything to it.

**Comment by Ali-Khamis** [comment]
@Nilegfx  changes done

File: modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.stories.tsx [hunk 7]
```diff
@@ -34,6 +34,6 @@ export const postViewHeader = Template.bind({});
 postViewHeader.args = {
   size: 'medium',
   variant: 'filled',
-  name: 'Marzouk el akta3',
```
Comment (Nilegfx) [inline modules/shared/components/molecules/PostViewHeader/UserInfo/PostViewHeader.stories.tsx]: It was nice. I didn't mean to change it. I like writing funny things in the tests. please keep it 😂

**Review by ahmedkhitaby — APPROVED** [review]
LGTM

**Pull request merged**